### PR TITLE
Fix #393: Improve the documentation structure

### LINKED ISSUE #393: Improve the documentation structure
There is a lot of work we need to do to make the documentation a bit more comprehensive. Most of the information dates back to the original protocol version. Since then, we introuduced plenty new features and we understood some concepts a bit better from the customer's point of view.

### PR DESCRIPTION
**Pull request by petrdvorak**

### CODE REVIEW

**Review by hvge — CHANGES_REQUESTED** [review]

File: docs/Activation-via-Activation-Code.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+- **Master Front-End Application Client** - An application that initiates the activation process by displaying the activation code to help the PowerAuth Client start the key exchange flow. Example of Master Front-End Application can be an Internet banking.
+- **Master Front-End Application Back-End** - An application that initiates the activation process by displaying the activation code to help the PowerAuth Client start the key exchange flow. Example of Master Front-End Application can be an Internet banking.
+- **PowerAuth Server** - A server application hidden deep in secure infrastructure, stores activation records, or verifies the request signatures. This application provides services for Intermediate Server Application to implement the PowerAuth protocol. An example of a PowerAuth Server is a bank identity management system.
```
[the remark this answers is not in the document]
Comment (petrdvorak) [reply docs/Activation-via-Activation-Code.md]: OK

File: docs/Activation.md [hunk 2]
```diff
@@ -1,157 +1,64 @@
...
 
-1. PowerAuth Server receives `ACTIVATION_CODE` and `ACTIVATION_DATA` from Intermediate Server Application. The `ACTIVATION_CODE` identifies the record for a pending activation. If the record is unknown, then server returns a generic error.
+You can read more about specific activation types in dedicated chapters:
 
-1. PowerAuth Server decrypts `ACTIVATION_DATA` with using an application scoped ECIES (level 2, `sh1="/pa/activation"`) and stores `KEY_DEVICE_PUBLIC` at given record.
+- [Activation via Activation Code](./Activation-via-Activation-Code.md)
+- [Activation via Recovery Code](./Activation-Recovery.md)
+- [Activation via Custom Credentials](#activation-via-custom-credentials)
```
[the remark this answers is not in the document]
Comment (petrdvorak) [reply docs/Activation.md]: Fixed

**Comment by petrdvorak** [comment]
I think this is good enough as the first rework of the structure. We can always adjust the documentation and add chapters, but I recommend merging what we have now.

**Review by hvge — APPROVED** [review]
Looks OK, I just found out that one image is a bit downgraded in resolution.

**Pull request merged**